### PR TITLE
Added a method on the NVDAObject class called objectFromPointRedirect

### LINKED ISSUE #16788: Add a getObjectFromPointRedirect method on the NVDAObject class
### Describe the solution you'd like
Add a getObjectFromPointRedirect method on the NVDAObject class that takes x and y coordinates as parameters and returns None in the default implementation. Call this method inside the objectFromPoint method, and send the x and y coordinates as parameters. If it returns an object, treat that object as the new objectFromPoint. In this way, add-ons can implement proper mouse reading and touch support for objects that has a location that differs from the location provided by the API class. This is useful if you for example are implementing support for a custom list control that don't support any accessibillity apis, and have to rely on win32 messages. The objectFromPoint method will normaly just return the root object. You could then implement getObjectFromPointRedirect on it to instead return the list item in that exact location.
### Describe alternatives you've considered
None

### PR DESCRIPTION
**Pull request by Emil-18**
### Link to issue number:
fixes #16788
### Summary of the issue:
It is impossible to redirect what object NVDA retrieves  from on screen coordinates. This makes it dificult to implement mouse and touch support for objects that has a location that differs from the location provided by the API class.
### Description of user facing changes
None
### Description of development approach
A new method called objectFromPointRedirect has been added to the NVDAObject class. This method is called from objectFromPoint, and given the x and y parameters. If it returns an NVDAObject, that object will be returned by objectFromPoint
### Testing strategy:
Tested by always returning the desktop object. It worked as it should.
### Known issues with pull request:
None
### Code Review Checklist:

- [x] Documentation:
  - Change log entry
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] API is compatible with existing add-ons.
- [x] Security precautions taken.

## Summary by CodeRabbit

- **New Features**
	- Added a method to redirect NVDA to another object based on screen coordinates.

- **Improvements**
	- Logging enhanced to include executable architecture at startup.
	- Improved task scheduling with a new module utilizing the `schedule` pip module.

### CODE REVIEW

File: source/NVDAObjects/__init__.py [hunk 1]
```diff
@@ -308,6 +308,17 @@ def findOverlayClasses(self, clsList: typing.List[typing.Type["NVDAObject"]]) ->
 
 	beTransparentToMouse=False #:If true then NVDA will never consider the mouse to be on this object, rather it will be on an ancestor.
 
+	def objectFromPointRedirect(self, x, y):
```
Comment (LeonarddeR) [inline source/NVDAObjects/__init__.py:311]: Please use type hints:
```suggestion
	def objectFromPointRedirect(self, x: int, y: int) -> NVDAObject | None:
```

File: source/NVDAObjects/__init__.py [hunk 2]
```diff
@@ -308,6 +308,17 @@ def findOverlayClasses(self, clsList: typing.List[typing.Type["NVDAObject"]]) ->
...
+		"""Redirects NVDA to another object if this object is retrieved from on screen coordinates.
+		@param x: the x coordinate.
```
Comment (LeonarddeR) [inline source/NVDAObjects/__init__.py:313]: The current style guide uses `:param:` I believe:
```suggestion
		:param x: the x coordinate.
```

File: source/NVDAObjects/__init__.py [hunk 3]
```diff
@@ -308,6 +308,17 @@ def findOverlayClasses(self, clsList: typing.List[typing.Type["NVDAObject"]]) ->
...
+		@type x: int
```
Comment (LeonarddeR) [inline source/NVDAObjects/__init__.py:314]: These are not necessary when annotating the function def correctly
```suggestion
```

**Review by LeonarddeR — CHANGES_REQUESTED** [review]

**Comment by Emil-18** [comment]
@LeonarddeR the line def objectFromPointRedirect(self, x: int, y: int) -> NVDAObject | None: gave an error since NVDAObject is not defined yet. Should I enclose the entire type hint in a string?

**Comment by LeonarddeR** [comment]
you should indeed store "NVDAObject" in a String probably.

**Comment by Emil-18** [comment]
@LeonarddeR Ok. Do you know how I can check what causes the linting issues, as the flake8.txt file is no longer present?

File: source/NVDAObjects/__init__.py [hunk 4]
```diff
@@ -316,6 +316,17 @@ def findOverlayClasses(self, clsList: typing.List[typing.Type["NVDAObject"]]) ->
...
 
+	def objectFromPointRedirect(self, x: int, y: int) -> "NVDAObject | None":
+		"""Redirects NVDA to another object if this object is retrieved from on screen coordinates.
+		:param x: the x coordinate.
+		:type x: int
+		:param y: the y coordinate.
+		:type y: int
+		:return: The object that NVDA should be redirected to.
+		:rtype: L{NVDAObject}
+		"""
```
Comment (seanbudd) [inline source/NVDAObjects/__init__.py:327]: The types are in the function declaration type hints, we don't add them to the docstring

```suggestion
		"""Redirects NVDA to another object if this object is retrieved from on-screen coordinates.
		:param x: the x coordinate.
		:param y: the y coordinate.
		:return: The object that NVDA should be redirected to.
		"""
```

File: source/NVDAObjects/__init__.py [hunk 5]
```diff
@@ -328,7 +339,13 @@ def objectFromPoint(x, y):
 		"""
```
Comment (seanbudd) [inline source/NVDAObjects/__init__.py:339]: as you've rewritten this function, could you update this function header to:
- move the type hints from the docstring into the function header
- update the docstring to use sphinx syntax

File: user_docs/en/changes.md [hunk 6]
```diff
@@ -109,6 +109,7 @@ It is especially useful to read the error location markers in tracebacks. (#1632
   * Table metadata can be added to an optional `brailleTables` section in the add-on manifest or to a `.ini` file with the same format found in the brailleTables subdirectory of the scratchpad directory.
   * Please consult the [braille translation tables section in the developer guide](https://www.nvaccess.org/files/nvda/documentation/developerGuide.html#BrailleTables) for more details.
 * When a `gainFocus` event is queued with an object that has a valid `focusRedirect` property, the object pointed to by the `focusRedirect` property is now held by `eventHandler.lastQueuedFocusObject`, rather than the originally queued object. (#15843)
+* It is now possible to redirect objects retrieved from on screen coordinates, by using the NVDAObject.objectFromPointRedirect method. (#16788, @Emil-18)
```
Comment (seanbudd) [inline user_docs/en/changes.md:112]: This has been added to the wrong release, please move to 2024.4

Also:

```suggestion
* It is now possible to redirect objects retrieved from on-screen coordinates, by using the `NVDAObject.objectFromPointRedirect` method. (#16788, @Emil-18)
```

**Review by seanbudd — CHANGES_REQUESTED** [review]
Thanks @Emil-18

**Review by seanbudd — APPROVED** [review]

**Pull request merged**